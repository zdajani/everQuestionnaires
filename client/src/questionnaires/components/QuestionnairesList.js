import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'
import _ from 'lodash';
import './styles/Questionnaires.css'; 

const QuestionnairesList = ({ questionnaires, url, isAdminPage }) => {
  const renderViewAnswersButtons = (questionnaire) => (
    <div className="questionnaires-list-buttons">
      <Link to={`${url}/${questionnaire.id}`} className="btn btn-info btn-lg">
        View Responses
      </Link>
    </div>
  )
  
  const renderTakeQuestionnaireButton = (questionnaire) => (
    <div className="questionnaires-list-buttons">
      <Link to={`${url}/${questionnaire.id}`} className="btn btn-info btn-lg">
        Take Questionnaire
      </Link>
    </div>
  )


  
  const renderQuestionnaires = () =>  {
    return _.map(questionnaires, questionnaire => {
      return (
        <div key={questionnaire.id} className="card">
          <Link to={`${url}/${questionnaire.id}`}>
            <h4 className="card-title"> {questionnaire.title} Questionnaire</h4>
          </Link>
            { isAdminPage ? renderViewAnswersButtons(questionnaire) : renderTakeQuestionnaireButton(questionnaire) }
        </div>
      )
    });
  }
  
  return (
    <div className="album text-muted">
        <div className="container">
          <div className="row questionnaires">
            { renderQuestionnaires() }
          </div>
        </div>
    </div>
  );
}

QuestionnairesList.propTypes = {
  questionnaires: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
  }).isRequired).isRequired,
  url: PropTypes.string.isRequired
};

export default QuestionnairesList;