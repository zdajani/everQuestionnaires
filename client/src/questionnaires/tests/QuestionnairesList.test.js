import React from 'react';
import { shallow } from 'enzyme';
import QuestionnairesList from '../components/QuestionnairesList';
import { formattedData } from './testData';
import toJson from 'enzyme-to-json';

//initialize with props if none are given

const setup = (questionnaires =formattedData, url = '/questionnaires' ) => {
  const component = shallow(
    <QuestionnairesList questionnaires={questionnaires} url={url}   />
  );
  
  return {
    component
  };
}

describe('QuestionnairesList component', () => {
  const { component } = setup();
  
  it('renders self with questionnaires props', () => {
    expect(component.instance().props).toEqual(
      {questionnaires: formattedData, url:'/questionnaires'}
    );
     
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders admin urls if admin url is given as prop', () => {
    const { component } = setup(formattedData, '/admin');
    expect(component.instance().props).toEqual(
      {questionnaires: formattedData, url:'/admin'}
    );
     
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders a list of questionnaire links', () => {
    expect(toJson(component.find('ul'))).toMatchSnapshot();
  });
});