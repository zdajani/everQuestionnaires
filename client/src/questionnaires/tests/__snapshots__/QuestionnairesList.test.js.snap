// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnairesList component renders a list of questionnaire links 1`] = `undefined`;

exports[`QuestionnairesList component renders admin urls if admin url is given as prop 1`] = `
<div
  className="album text-muted"
>
  <div
    className="container"
  >
    <div
      className="row questionnaires"
    >
      <div
        className="card"
      >
        <Link
          replace={false}
          to="/admin/1"
        >
          <h4
            className="card-title"
          >
             
            Lord of the Rings
             Questionnaire
          </h4>
        </Link>
        <div
          className="questionnaires-list-buttons"
        >
          <Link
            className="btn btn-info btn-lg"
            replace={false}
            to="/admin/1"
          >
            Take Questionnaire
          </Link>
        </div>
      </div>
      <div
        className="card"
      >
        <Link
          replace={false}
          to="/admin/2"
        >
          <h4
            className="card-title"
          >
             
            Lord of the Flies
             Questionnaire
          </h4>
        </Link>
        <div
          className="questionnaires-list-buttons"
        >
          <Link
            className="btn btn-info btn-lg"
            replace={false}
            to="/admin/2"
          >
            Take Questionnaire
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`QuestionnairesList component renders self with questionnaires props 1`] = `
<div
  className="album text-muted"
>
  <div
    className="container"
  >
    <div
      className="row questionnaires"
    >
      <div
        className="card"
      >
        <Link
          replace={false}
          to="/questionnaires/1"
        >
          <h4
            className="card-title"
          >
             
            Lord of the Rings
             Questionnaire
          </h4>
        </Link>
        <div
          className="questionnaires-list-buttons"
        >
          <Link
            className="btn btn-info btn-lg"
            replace={false}
            to="/questionnaires/1"
          >
            Take Questionnaire
          </Link>
        </div>
      </div>
      <div
        className="card"
      >
        <Link
          replace={false}
          to="/questionnaires/2"
        >
          <h4
            className="card-title"
          >
             
            Lord of the Flies
             Questionnaire
          </h4>
        </Link>
        <div
          className="questionnaires-list-buttons"
        >
          <Link
            className="btn btn-info btn-lg"
            replace={false}
            to="/questionnaires/2"
          >
            Take Questionnaire
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
